pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.cpp

```diff
@@ -17,3 +17,19 @@
 // explicit instantiation
 template class Pennylane::StateVectorKokkos<float>;
 template class Pennylane::StateVectorKokkos<double>;
+
+std::string repr_InitArguments(const Kokkos::InitArguments &a) {
+    std::string str;
+    str = "<example.InitArguments with";
+    str += "\n num_threads = " + std::to_string(a.num_threads);
+    str += "\n num_numa = " + std::to_string(a.num_numa);
+    str += "\n device_id = " + std::to_string(a.device_id);
+    str += "\n ndevices = " + std::to_string(a.ndevices);
+    str += "\n skip_device = " + std::to_string(a.skip_device);
+    str += "\n disable_warnings = " + std::to_string(a.disable_warnings) += ">";
+    return str;
+}
+
+void print_InitArguments(const Kokkos::InitArguments &a) {
+    std::cout << repr_InitArguments(a) << '\n';
+}
```

Comment:
Probably not. I am not familiar with the project, but `pennylane_lightning_kokkos/src/util/Util.hpp` may be one place to put it since the doc says "Contains uncategorised utility functions." Then #include "Util.hpp" where necessary.

Reply:
> Also, did you consider defining the streaming operator for the `Kokkos::InitArguments` struct?

I have not considered it. Something like 
```
ostream &operator<<(ostream &os, const Kokkos::InitArguments &a)
{
   os << std::cout << repr_InitArguments(a);
   return os;
}
```
? Could work for a class, but I'm unsure if possible for a struct.

Reply:
Since a CPP struct and class are the same, except a struct defaults to all members being public, the streaming operator would be the preferred option here.

Reply:
I added the streaming operator and related functions in `pennylane_lightning_kokkos/src/util/KokkosUtil.hpp`.